packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -142,7 +142,9 @@ export function useAsyncData<
   const nuxt = useNuxtApp()
 
   const getCachedData = () => nuxt.isHydrating ? nuxt.payload.data[key] : nuxt.static.data[key]
-  const hasCachedData = () => getCachedData() !== undefined
+  const hasCachedData = () => nuxt.isHydrating
+    ? (nuxt.payload.data[key] !== null && nuxt.payload.data[key] !== undefined)
+    : nuxt.static.data[key] !== undefined
```

Comment:
The issue here seems to be that when there is an _error_ the key is set to `null`.